Find tests recursively

r0man opened the pull request:
Hello @kirbyfan64,

I'm just starting to explore Hy and was looking for a unit test framework and found yours. However I have the need to run tests in nested directories. This patch changes the find-tests function to find those tests as well. I changed the signature of find-tests so it is a bit more flexible, but mainly for testing purpose.

Would you like to merge this?

Thanks, Roman.

kirbyfan64 commented on hytest.hy:
```diff
@@ -326,21 +326,15 @@
+  (for [p (find-tests (getcwd))]
     (mods.append
-      (import-file-to-module (get (path.splitext (path.basename p)) 0) p))))
+     (import-file-to-module (get (path.splitext (path.basename p)) 0) p))))
```
I don't think the removal of the space here was intentional... ;)

r0man replied:
Oops, yes, that was my Emacs setup doing auto indentation. Fixed.

kirbyfan64 commented on test_hytest.hy:
```diff
@@ -87,3 +90,22 @@
+
+(defn touch [&rest parts]
+  "Join the path `parts` and touch the resulting filename."
+  (.close (open (apply path.join (list parts)) "a")))
```
Does this not work without the `(list parts)`? I thought it was supposed to.

r0man replied:
Yes, you are right, it does work without the call to list. I removed it.

kirbyfan64 commented:
Thanks! I pointed out two *really* minor things, though, if you're unable fix them, I'll just merge this as-is and fix them myself.

r0man commented:
@kirbyfan64 I addressed your comments. The build hangs since 20 mins on Travis. I'm out now but will take a look tomorrow if you have more suggestions.

r0man commented:
Ok, the tests just finished.

kirbyfan64 commented:
@r0man Thanks again! I've got to take a look at speeding up the Travis runs...
